models/snowplow_fractribution_paths_to_conversion.sql

```diff
@@ -0,0 +1,53 @@
+{{
+  config(
+    sql_header=snowplow_utils.set_query_tag(var('snowplow__query_tag', 'snowplow_dbt'))
+  )
+}}
+
+-- Requires macro trim_long_path
+
+with string_aggs as (
+
+  select distinct
+    c.customer_id,
+    c.conversion_tstamp,
+    revenue,
+    {{ snowplow_utils.get_string_agg('channel', 's', separator=' > ', sort_numeric=false, order_by_column='visit_start_tstamp', partition_by_columns='c.customer_id', order_by_column_prefix='s') }} as path,
+    {{ snowplow_utils.get_string_agg('channel', 's', separator=' > ', sort_numeric=false, order_by_column='visit_start_tstamp', partition_by_columns='c.customer_id', order_by_column_prefix='s') }} as transformed_path
+
+  from {{ ref('snowplow_fractribution_conversions_by_customer_id') }} c
+
+  left join {{ ref('snowplow_fractribution_sessions_by_customer_id') }} s
+  on c.customer_id = s.customer_id
+    and {{ datediff('s.visit_start_tstamp', 'c.conversion_tstamp', 'day') }}  >= 0
+    and {{ datediff('s.visit_start_tstamp', 'c.conversion_tstamp', 'day') }} <= {{ var('path_lookback_days') }}
+
+{% if target.type in ['snowflake', 'bigquery'] -%}
+  group by 1,2,3
+{%- endif %}
```

Comment:
why don't we group for databricks?

Reply:
Because the workaround solution for the utils' databricks__get_string_agg (to be able to sort based on another column) involves using a non-aggregate window function, whereas for the rest we have aggregate functions like string_agg for bigquery.

Reply:
Does that not mean we need to do a `distinct` for databricks? I also absolutely hate that we have different return types for the same function on the same warehouse, let alone across warehouses...

Reply:
I guess also for future-proofing it should be `if target.type not in ['databricks', 'spark']` ?

Reply:
Yes, that's why I added 'distinct', it is for databricks only. Definitely not ideal but that's the only solution for sorting based on another column.